### ISSUE TITLE
Stress test network

### ISSUE THREAD

**Issue by ilblackdragon** [near/nearcore#2423]
Bombard the node with correct block data, but with intensity of DDOS.
 Make sure node protects against it by blocking the incoming connection.

**Comment by mfornet** [comment]
Some protection for DDOS already exists, with `is_abusive` heuristic which detect number of message per minute received. Currently we have it disabled, since in practice our current implementation is consuming a lot of bandwidth and most of the peers were being kicked out.

We can try to do something smarter such as counting repeated messages (or useless messages).

**Comment by bowenwang1996** [comment]
> Currently we have it disabled, since in practice our current implementation is consuming a lot of bandwidth and most of the peers were being kicked out.

Can we do a better estimation or is that just not feasible?

**Comment by bowenwang1996** [comment]
> We should differentiate per type of message and probably tag some message as "irrelevant" and identify and kick out spammers by the amount of bytes per minute send on irrelevant messages.

Makes sense. I think we can have limit on the number of messages per type instead of looking at the sizes alone.
